PurchasesCoreSwift/Public/Purchases.swift

```diff
@@ -0,0 +1,1275 @@
+        set { StoreKitWrapper.simulatesAskToBuyInSandbox = newValue }
+    }
+
+    /**
+     * Enable debug logging. Useful for debugging issues with the lovely team @RevenueCat.
+     */
+    @objc static var debugLogsEnabled: Bool {
+        get { logLevel == .debug }
+        set { logLevel = newValue ? .debug : .info }
+    }
+
+    /**
+     * Set a custom log handler for redirecting logs to your own logging system.
+     * By default, this sends Info, Warn, and Error messages. If you wish to receive Debug level messages,
+     * you must enable debug logs.
+     */
+    @objc static var logHandler: (LogLevel, String) -> Void {
+        get { Logger.logHandler }
+        set { Logger.logHandler = newValue }
+    }
+
+    /// Current version of the Purchases framework.
+    @objc static var frameworkVersion: String { SystemInfo.frameworkVersion }
+
+    @objc var allowSharingAppStoreAccount: Bool {
+        get { purchasesOrchestrator.allowSharingAppStoreAccount }
+        set { purchasesOrchestrator.allowSharingAppStoreAccount = newValue }
+    }
+
+    @objc var finishTransactions: Bool {
+        get { systemInfo.finishTransactions }
+        set { systemInfo.finishTransactions = newValue }
+    }
+
+    @objc static var sharedPurchases: Purchases {
+        if let purchases = purchases {
+            return purchases
+        }
+
+        notConfiguredAssertionFunction()
+
+        // This will only be returned during testing. #hack for ARM64 testing fatalError()
+        return PurchasesTestStandIn(apiKey: "StandIn", appUserID: nil)
+    }
+
+    static var shared: Purchases { sharedPurchases }
+    @objc static var isConfigured: Bool { purchases != nil }
+
+    private static var purchases: Purchases?
+
+    private let requestFetcher: StoreKitRequestFetcher
+    private let productsManager: ProductsManager
+    private let receiptFetcher: ReceiptFetcher
+    private let backend: Backend
+    private let storeKitWrapper: StoreKitWrapper
+    private let notificationCenter: NotificationCenter
+    private let attributionFetcher: AttributionFetcher
+    private let attributionPoster: AttributionPoster
+    private let offeringsFactory: OfferingsFactory
+    private let deviceCache: DeviceCache
+    private let identityManager: IdentityManager
+    private let systemInfo: SystemInfo
+    private let introEligibilityCalculator: IntroEligibilityCalculator
+    private let receiptParser: ReceiptParser
+    private let purchaserInfoManager: PurchaserInfoManager
+    private let offeringsManager: OfferingsManager
+    private let purchasesOrchestrator: PurchasesOrchestrator
+
+    fileprivate static let initLock = NSLock()
+
+    static var notConfiguredAssertionFunction: () -> Void = { fatalError(Strings.purchase.purchases_nil) }
+
+    /**
+     * Indicates whether the user is allowed to make payments.
+     */
+    @objc public static func canMakePayments() -> Bool { SKPaymentQueue.canMakePayments() }
```

Comment:
style: this should with the other public computed variables

Reply:
💯  ✅